api/src/test/java/org/pmiops/workbench/interceptors/AuthInterceptorTest.java

```diff
@@ -280,4 +286,29 @@ public void authorityCheckPermitsWhenUserHasAuthority() throws Exception {
     Method apiControllerMethod = FakeApiController.class.getMethod("handle");
     assertThat(interceptor.hasRequiredAuthority(apiControllerMethod, user)).isTrue();
   }
+
+  @Test
+  public void preHandle_apiBaseUrlNotMatch_forbidden() throws Exception {
+    when(mockHandler.getMethod()).thenReturn(getTestMethod());
+    when(mockRequest.getMethod()).thenReturn(HttpMethods.GET);
+
+    when(mockRequest.getRequestURL()).thenReturn(new StringBuffer("domain"));
+
+    assertThat(interceptor.preHandle(mockRequest, mockResponse, mockHandler)).isFalse();
+    verify(mockResponse).sendError(HttpServletResponse.SC_FORBIDDEN);
+  }
+
+  @Test
+  public void preHandle_apiBaseUrlNotMatch_cloudTask_allowed() throws Exception {
```

Comment:
test for Cron as well?

Reply:
We are not adding checks for Cron job. Cron job does not have user bearer token and aou_auth, it returns true before reaching to the new check: https://github.com/all-of-us/workbench/blob/7a37a332a0f6c8e833af46f217e89a880dad8086/api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java#L105

Seems our existing `CronInterceptorTest` has good coverage already